bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/regexp/TreeBuilder.java

```diff
@@ -197,6 +259,28 @@ private String readRegUnicodeGeneralCategory() {
         throw new BallerinaException(getErrorMsg(nextToken));
     }
 
+    /**
+     * Read close brace.
+     *
+     * @return '}' character
+     */
+    private String readCloseBrace() {
+        try {
+            Token nextToken = peek();
+            if (nextToken.kind == TokenKind.CLOSE_BRACE_TOKEN) {
+                return consume().value;
+            }
+        } catch (BallerinaException ignored) {
```

Comment:
Since we are ignoring the exception here, the actual reason will be hidden and will be hard to debug when an issue arises, isn't it?

Reply:
Yes. I'll refactor the method.

Reply:
Yes. But since this is runtime parsing I think it is ok to ignore that error an throw the `missing close brace token error`.